torch/distributed/run.py

```diff
@@ -781,7 +781,7 @@ def run_script_path(training_script: str, *training_script_args: str):
 def run(args):
     if args.standalone:
         args.rdzv_backend = "c10d"
-        args.rdzv_endpoint = "localhost:29400"
+        args.rdzv_endpoint = "localhost:0"
```

Comment:
Can you add a test using `--standalone` to confirm that this is still working when world_size > 1?

This looks like a good place to add:
https://github.com/pytorch/pytorch/blob/42897e81270ddad2449504d4fe38a673f8a95a07/test/distributed/launcher/run_test.py

Reply:
Thanks! Hm, looking at https://github.com/pytorch/pytorch/blob/42897e81270ddad2449504d4fe38a673f8a95a07/test/distributed/launcher/run_test.py#L374 there's a test for nnodes=1 and nproc=4 already

Is there another config I should check?

Reply:
This is interesting, thanks for checking. I am suspicious of whether these tests are even being run. cc @kurman and PyTorch R2P team, are these tests included in PyTorch CI?

My hunch was this wouldn't work when there is more than 1 process. This is because I thought each process would connect to a separate port (wildcard port) and they would not be able to rendezvous, thus a static port is needed. I'll make a post in the r2p team wp group since they would know more.

Reply:
Also, maybe related:
- https://github.com/pytorch/pytorch/issues/44544

Reply:
Hm, as a quick sanity test to make sure things actually worked I also tried running the following snippet with production pytorch -- 

```
(/home/kunalb/dev/torch/testdist) -- 18.15.58 - ~/dev/torch -
   cat test.sh
#!/bin/bash
echo $MASTER_PORT " -- " $MASTER_ADDR
(/home/kunalb/dev/torch/testdist) -- 18.16.00 - ~/dev/torch -
   torchrun --nnodes 1 --no-python --nproc_per_node 8 --rdzv_backend=c10d --rdzv_endpoint=localhost:0  ./test.sh
master_addr is only used for static rdzv_backend and when rdzv_endpoint is not specified.
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
*****************************************
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
38327  --  kunalb-fedora-PC2BDTCH
```

It does look like the port gets propagated correctly, looking through the code to make sure I can explain why.

Reply:
Sounds good @kunalb , thank you for running that sanity check! If that's the case, then I will follow up on the missing CI coverage separately and not have it block this diff